src/Core/Framework/Api/Controller/AuthController.php

```diff
@@ -49,7 +50,11 @@ public function token(Request $request): Response
         $psr7Request = $this->psrHttpFactory->createRequest($request);
         $psr7Response = $this->psrHttpFactory->createResponse($response);
 
-        $response = $this->authorizationServer->respondToAccessTokenRequest($psr7Request, $psr7Response);
```

Comment:
The 400 error is spec compliant, therefore i see no need to change it here, for details take a look at the PRs where this was changed and discussed in the OAuth lib:
https://github.com/thephpleague/oauth2-server/pull/967
https://github.com/thephpleague/oauth2-server/pull/1230

Ans also look at this SO post also referenced in the PRs: https://stackoverflow.com/questions/22586825/oauth-2-0-why-does-the-authorization-server-return-400-instead-of-401-when-the